### PR TITLE
[32-74] `FlowSplitWhenSpec`

### PR DESCRIPTION
**Pull request by eaba**
## Changes

- Change `AssertAllStagesStopped` to `AssertAllStagesStoppedAsync`

### CODE REVIEW

File: src/core/Akka.Streams.Tests/Dsl/FlowSplitWhenSpec.cs [hunk 1]
```diff
@@ -85,108 +90,113 @@ public StreamPuppet(IPublisher<int> p, TestKitBase kit)
...
+            await this.AssertAllStagesStoppedAsync(async() => {
+                await WithSubstreamsSupportAsync(elementCount: 4, 
+                    run: (masterSubscriber, masterSubscription, getSubFlow) =>                                                                                            
+                    {                                                                             
+                        var s1 = new StreamPuppet(getSubFlow()
+                            .RunWith(Sink.AsPublisher<int>(false), Materializer), this);                                                                             
+                        masterSubscriber.ExpectNoMsg(TimeSpan.FromMilliseconds(100));
+                                                                             
+                        s1.Request(2);                                                                             
+                        s1.ExpectNext(1);
```
Comment (Aaronontheweb) [inline src/core/Akka.Streams.Tests/Dsl/FlowSplitWhenSpec.cs]: Don't all of these need to be `async` calls @eaba ?

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Looks like there are some specs that aren't using the `async` methods yet

File: src/core/Akka.Streams.Tests/Dsl/FlowSplitWhenSpec.cs (hunk 1 above)
Comment (eaba) [reply src/core/Akka.Streams.Tests/Dsl/FlowSplitWhenSpec.cs]: Ok. Let me check it out

Comment (eaba) [reply src/core/Akka.Streams.Tests/Dsl/FlowSplitWhenSpec.cs]: Done

**Review by Aaronontheweb — DISMISSED** [review]
LGTM

**Comment by eaba** [comment]
This is the problem I'm having here `SplitWhen_must_work_in_the_happy_case` @Aaronontheweb 

image

**Comment by Aaronontheweb** [comment]
> This is the problem I'm having here `SplitWhen_must_work_in_the_happy_case` @Aaronontheweb
> 
> image

You'll need to find a way to work the problem

**Comment by Aaronontheweb** [comment]
There are errors with these tests but the test runner crashed while trying to report on the results. 

```
[xUnit.net 00:06:41.05]       Racy on Azure DevOps
  Passed Akka.Streams.Tests.Dsl.FlowSplitWhenSpec.SplitWhen_must_support_cancelling_the_master_stream [109 ms]
  Passed Akka.Streams.Tests.Dsl.FlowSplitWhenSpec.SplitWhen_must_fail_stream_if_substream_not_materialized_in_time [1 s]
  Skipped Akka.Streams.Tests.Dsl.FlowSplitWhenSpec.SplitWhen_must_resume_stream_when_splitWhen_function_throws [1 ms]
  Passed Akka.Streams.Tests.Dsl.FlowSplitWhenSpec.SplitWhen_must_pass_along_early_cancellation [124 ms]
  Passed Akka.Streams.Tests.Dsl.FlowSplitWhenSpec.SplitWhen_must_work_with_single_element_splits [107 ms]
  Skipped Akka.Streams.Tests.Dsl.FlowSplitWhenSpec.SplitWhen_must_fail_substream_if_materialized_twice [1 ms]
The active test run was aborted. Reason: Test host process crashed : Unhandled exception. Xunit.Sdk.TrueException: Failed: Expected no messages during 00:00:00.1000000, instead we received <TestSubscriber.OnError(Processor actor [[akka://FlowSplitWhenSpec-1140/user/StreamSupervisor-1753/Flow-0-0-unknown-operation#1873049784]] terminated abruptly)> from [akka://FlowSplitWhenSpec-1140/user/StreamSupervisor-1753/Flow-0-0-unknown-operation#1873049784] after 00:00:00.0157833
Expected: True
Actual:   False
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 132
   at Akka.TestKit.Xunit2.XunitAssertions.Fail(String format, Object[] args) in D:\a\1\s\src\contrib\testkits\Akka.TestKit.Xunit2\XunitAssertions.cs:line 26
   at Akka.TestKit.TestKitBase.InternalExpectNoMsgAsync(TimeSpan duration, CancellationToken cancellationToken) in D:\a\1\s\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 535
   at Akka.TestKit.TestKitBase.ExpectNoMsgAsync(TimeSpan duration, CancellationToken cancellationToken) in D:\a\1\s\src\core\Akka.TestKit\TestKitBase_Expect.cs:line 501
   at Akka.Streams.TestKit.TestSubscriber.ManualProbe`1.ExpectNoMsgAsync(TimeSpan remaining, CancellationToken cancellationToken) in D:\a\1\s\src\core\Akka.Streams.TestKit\TestSubscriber.cs:line 239
   at Akka.Streams.Tests.Dsl.FlowSplitWhenSpec.<SplitWhen_must_work_in_the_happy_case>b__6_1(ManualProbe`1 masterSubscriber, ISubscription masterSubscription, Func`1 getSubFlow) in D:\a\1\s\src\core\Akka.Streams.Tests\Dsl\FlowSplitWhenSpec.cs:line 96
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
```

It's the "happy case" spec that's failing again.

**Comment by eaba** [comment]
Yes, only local! The problem here `ExpectNoMsgAsync`s `TestSubscriber.OnError`

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**